src/omefiles.cpp

```diff
@@ -26,67 +26,139 @@
  * #L%
  */
 
-#include <Python.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <string>
+#include <vector>
+#include <cstdint>
+#include <ome/files/Version.h>
+#include <ome/files/in/OMETIFFReader.h>
+#include <ome/xml/meta/MetadataStore.h>
+#include <ome/xml/meta/OMEXMLMetadata.h>
+#include <ome/xml/model/enums/PixelType.h>
 
-#include "ometiffreader.h"
-#include "errors.h"
+namespace py = pybind11;
+using ome::files::in::OMETIFFReader;
 
-#if PY_MAJOR_VERSION >= 3
-#define PY3
-#define INIT_RETURN(V) return V;
-#else
-#define INIT_RETURN(V) return;
-#endif
-
-PyObject *OMEFilesPyError;
-
-
-static PyMethodDef OMEFilesMethods[] = {
-  {NULL}  /* Sentinel */
-};
-
-
-#ifdef PY3
-static struct PyModuleDef module_def = {
-  PyModuleDef_HEAD_INIT,
-  "_core",
-  "OME Files wrapper",
-  0,
-  OMEFilesMethods,
-  NULL,
-  NULL,
-  NULL,
-  NULL
-};
-#endif
-
-
-PyMODINIT_FUNC
-#ifdef PY3
-PyInit__core(void) {
-#else
-init_core(void) {
-#endif
-  PyObject *m;
-  PyOMETIFFReaderType.tp_new = PyType_GenericNew;
-  if (PyType_Ready(&PyOMETIFFReaderType) < 0) {
-    INIT_RETURN(NULL);;
-  }
-#ifdef PY3
-  m = PyModule_Create(&module_def);
-#else
-  m = Py_InitModule3("_core", OMEFilesMethods, "OME Files wrapper");
-#endif
-  if (!m) {
-    INIT_RETURN(NULL);;
-  }
-  Py_INCREF(&PyOMETIFFReaderType);
-  PyModule_AddObject(m, "OMETIFFReader", (PyObject *)&PyOMETIFFReaderType);
-  OMEFilesPyError = PyErr_NewException(const_cast<char*>("ome_files.Error"),
-                                       NULL, NULL);
-  if (OMEFilesPyError) {
-    Py_INCREF(OMEFilesPyError);
-    PyModule_AddObject(m, "Error", OMEFilesPyError);
-  }
-  INIT_RETURN(m);
+PYBIND11_PLUGIN(_core) {
+  py::module m("_core");
+  py::class_<ome::files::Version>(m, "Version")
+    .def(py::init<uint32_t, uint32_t, uint32_t, const std::string &>())
+    .def_readonly("major", &ome::files::Version::major)
+    .def_readonly("minor", &ome::files::Version::minor)
+    .def_readonly("patch", &ome::files::Version::patch)
+    .def_readonly("extra", &ome::files::Version::extra);
+  m.attr("version_info") = ome::files::release_version;
+  py::class_<OMETIFFReader>(m, "OMETIFFReader")
+    .def("__init__", [](OMETIFFReader &r) {
+	new (&r) OMETIFFReader();
+	std::shared_ptr<ome::xml::meta::MetadataStore> store(
+          std::make_shared<ome::xml::meta::OMEXMLMetadata>()
+        );
+	r.setMetadataStore(store);
+      })
+    .def("set_id", [](OMETIFFReader &r, std::string id) {
+	r.setId(id);
+      }, "Set the current file name.")
+    .def("set_series", [](OMETIFFReader &r, size_t series) {
+        r.setSeries(series);
+      }, "Set the active series.")
+    .def("get_series", [](OMETIFFReader &r) {
```

Comment:
The getters and non-modifying methods could probably use `const OMETIFFReader &r`?